src/pandapipes/component_models/pipe_component.py

```diff
@@ -101,31 +115,43 @@ def create_pit_branch_entries(cls, net, branch_pit):
-        set_entry_check_repeat(
-            pipe_pit, D, net[tbl].diameter_m.values, internal_pipe_number, has_internals)
-        set_entry_check_repeat(
-            pipe_pit, LC, net[tbl].loss_coefficient.values, internal_pipe_number, has_internals)
-
-        nan_mask = np.isnan(pipe_pit[:, TEXT])
-        pipe_pit[nan_mask, TEXT] = get_net_option(net, 'ambient_temperature')
-        pipe_pit[:, AREA] = pipe_pit[:, D] ** 2 * np.pi / 4
-        pipe_pit[:, MDOTINIT] *= pipe_pit[:, AREA] * get_fluid(net).get_density(NORMAL_TEMPERATURE)
+
+        pipe_pit, node_pit = super().create_pit_branch_entries(net, branch_pit)
+        if not get_net_option(net, "transient") or get_net_option(net, "simulation_time_step") == 0:
+            junction_idx_lookup = get_lookup(net, "node", "index")[
+                cls.get_connected_node_type().table_name()]
+            fn_col, tn_col = cls.from_to_node_cols()
+
+            from_nodes = junction_idx_lookup[net[cls.table_name()][fn_col].values]
+            to_nodes = junction_idx_lookup[net[cls.table_name()][tn_col].values]
+            internal_pipe_number = cls.get_internal_branch_number(net)
+            has_internals = np.any(internal_pipe_number > 1)
+            if has_internals:
+                internal_node_number = cls.get_internal_node_number(net)
+                node_ft_lookups = get_lookup(net, "node", "from_to")
+                pipe_nodes_from, pipe_nodes_to = node_ft_lookups[cls.internal_node_name()]
+                pipe_nodes_idx = np.arange(pipe_nodes_from, pipe_nodes_to)
+                insert_places = np.repeat(np.arange(len(from_nodes)), internal_node_number)
+                from_nodes = np.insert(from_nodes, insert_places + 1, pipe_nodes_idx)
+                to_nodes = np.insert(to_nodes, insert_places, pipe_nodes_idx)
+
+            pipe_pit[:, FROM_NODE] = from_nodes
+            pipe_pit[:, TO_NODE] = to_nodes
+            pipe_pit[:, TOUTINIT] = node_pit[to_nodes, TINIT_NODE]
+
+            tbl = cls.table_name()
+            set_entry_check_repeat(pipe_pit, LENGTH, net[tbl].length_km.values * 1000 / internal_pipe_number,
+                internal_pipe_number, has_internals)
+            set_entry_check_repeat(pipe_pit, K, net[tbl].k_mm.values / 1000, internal_pipe_number, has_internals)
+            set_entry_check_repeat(pipe_pit, ALPHA, net[tbl].u_w_per_m2k.values, internal_pipe_number, has_internals)
+            set_entry_check_repeat(pipe_pit, QEXT, net[tbl].qext_w.values, internal_pipe_number, has_internals)
+            set_entry_check_repeat(pipe_pit, TEXT, net[tbl].text_k.values, internal_pipe_number, has_internals)
+            nan_mask = np.isnan(pipe_pit[:, TEXT])
+            pipe_pit[nan_mask, TEXT] = get_net_option(net, 'ambient_temperature')
+            pipe_pit[:, AREA] = pipe_pit[:, D] ** 2 * np.pi / 4
+            pipe_pit[:, MDOTINIT] *= pipe_pit[:, AREA] * get_fluid(net).get_density(NORMAL_TEMPERATURE)
```

Comment:
I know this is not related to the PR, but I see this for the first time and think that it is prone for errors. Why do we do that?